### PR TITLE
New: add Nav component

### PR DESCRIPTION
**Pull request by lightbringer1991**
Add Nav component, with `borderless bsStyle`

This is a wrapper around `react-bootstrap` Nav component, with the addition of `borderless` style
Works well with `NavItem`, `NavBar`

after

EDIT: Updated to match platform styles

navtav
navtavb

### CODE REVIEW

**Comment by vinteo** [comment]
how is this different from tabs?

**Comment by lightbringer1991** [comment]
@vinteo it doesn't have any tab content body, just the navigation bar with a bunch of buttons

File: docs/examples/NavigationExample.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+class NavigationExample extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeKey: 1,
+    };
+
+    this.onSelect = this.onSelect.bind(this);
```
Comment (omgaz) [inline docs/examples/NavigationExample.jsx]: https://adslot.atlassian.net/wiki/spaces/APT/pages/178225251/React

Can we use the class arrow functions. They're supported in Adslot UI through babel preset env.

File: docs/examples/NavigationExample.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  onSelect(eventKey) {
+    this.setState({ activeKey: eventKey });
+  }
+
+  render() {
+    return (
+      <Nav bsStyle="borderless" activeKey={this.state.activeKey} onSelect={this.onSelect}>
+        <NavItem eventKey={1} href="#">NavItem 1 content</NavItem>
+        <NavItem eventKey={2} title="Item">NavItem 2 content</NavItem>
+        <NavItem eventKey={3} disabled>NavItem 3 content</NavItem>
```
Comment (omgaz) [inline docs/examples/NavigationExample.jsx]: Dashboard
Reports

And we can remove the disabled one from the example.

File: src/components/adslot-ui/Navigation/index.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+}
+
+NavigationComponent.propTypes = _.assign({}, Nav.propTypes, {
+  bsStyle: PropTypes.oneOf(['tabs', 'pills', 'borderless']),
+  barPosition: PropTypes.oneOf(['top', 'bottom']),
+});
+
+NavigationComponent.defaultProps = {
+  bsStyle: 'pills',
```
Comment (omgaz) [inline src/components/adslot-ui/Navigation/index.jsx]: We should only be supporting borderless, so there shouldn't be a prop-type for this.

File: src/components/adslot-ui/Navigation/styles.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+.nav-borderless {
+  display: flex;
+  flex-direction: row;
+  font-size: $font-size-header;
+  color: $navbar-text-color;
+  font-weight: $font-weight-bold;
+
+  li {
+    list-style-type: none;
+    flex: 0 1 auto;
```
Comment (omgaz) [inline src/components/adslot-ui/Navigation/styles.scss]: Can you check this in FF and IE. Sometimes 0 can do weird things in IE and sometimes auto can do weird things in FF.

File: src/components/adslot-ui/Navigation/styles.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      a {
+        color: $color-text-disabled;
+      }
+    }
+  }
+
+  &.top-bar {
+    li {
+      .active {
+        border-top: 4px solid $color-primary-selected;
```
Comment (omgaz) [inline src/components/adslot-ui/Navigation/styles.scss]: The default for buttons is grey, can we use the same default highlight here with the intention that it is overridden by the implementation of the feature?

**Comment by lightbringer1991** [comment]
LGTM, can you remove need to remove the `package.lock` file?

**Comment by omgaz** [comment]
@lightbringer1991 could do, but they're just point releases from an `npm i --no-optional` so this is valid

**Comment by lightbringer1991** [comment]
@omgaz please merge when possible, thank you

**Pull request merged**